app/core/models.py

```diff
@@ -0,0 +1,31 @@
+from django.db import models
+
+
+class Recipe(models.Model):
+    """Recipe object"""
+    name = models.CharField(max_length=255)
+    description = models.CharField(max_length=255)
+
+    def create(self, validated_data):
+        ingredients = validated_data.pop('ingredients')
+        recipe = Recipe.objects.create(**validated_data)
+        if ingredients:
+            for ingredient in ingredients:
+                Ingredient.objects.create(recipe=recipe, **ingredient)
+        return recipe
+
+    def __str__(self):
+        return self.title
+
+
+class Ingredient(models.Model):
+    """Ingredient to be used in a recipe"""
+    name = models.CharField(max_length=255)
```

Comment:
An extra would be making the name unique (check https://docs.djangoproject.com/en/3.2/ref/models/fields/#unique .. though this is not case sensitive. Anyway, mentioning it for the sake of completeness)

Reply:
Are we sure we want to do this? The requirements for the task stated: `Assume a given ingredient belongs only to one recipe, even if that means multiple Ingredient instances with the exact same name.`

Having multiple ingredients of the same name, e.g. `cheese` in both `Pizza` and `Cheeseburger` would mean that if, for example, we deleted the `Pizza` recipe, then the cascade delete would only delete the `cheese` from the `Pizza` recipe, but not from the `Cheeseburger` as well.

So I don't think we want the ingredient name to be unique here.

Reply:
Oh, you are totally right :sweat_smile: